### ISSUE TITLE
Round accuracy is a weighted mean over label-skewed local shards

### ISSUE THREAD

**Issue by ajbarea** [ajbarea/phalanx-fl#104]
Every client evaluates on a holdout carved from its own partition. `phalanx/task.py` `load_data`:

```python
partition = _fds.load_partition(partition_id)
split = partition.train_test_split(test_size=0.2, seed=42)
```

`FedAvg.aggregate_evaluate` then takes a `num-examples`-weighted mean of those per-client scores, and that mean is what lands in `fl.accuracy`, `fl.round.accuracy`, and the run manifest.

Under `partitioner = "dirichlet"` the holdout inherits its partition's label skew, so a client whose partition is mostly one label scores well by predicting that label. The reported accuracy therefore measures partition-local agreement, and its value moves with `dirichlet-alpha` for reasons unrelated to how well the federated adapters generalise. Two runs at different alphas are not comparable on this number, and neither is a run against a centralized baseline.

This bears directly on a claim in `ROADMAP.md`:

> The round-2 Dirichlet accuracy collapse is genuine non-IID dynamics, not an aggregation bug. The IID control improves monotonically, which is what rules the bug out.

The IID control is the right instinct, but both arms are measured on local shards, so the control does not isolate what it is meant to. The finding may well hold — it is not currently evidenced by a number that can carry it.

## Recommended fix

Hold out a shared split once, before partitioning, and evaluate it server-side:

- Load the dataset's `test` split (IMDB ships one) as a global evaluation set, untouched by the partitioner.
- Add a server-side evaluation hook on `ObservableFedAvg` that scores the aggregated adapters on it each round, and emit it as a distinct metric — `fl.round.global_accuracy` — rather than overwriting the federated figure.
- Keep the per-client number too. Both are meaningful; the pair is what shows skew-driven divergence, which suits the observability angle.

Then re-run the IID and Dirichlet arms and restate the ROADMAP finding against the global metric, or retire it.

Scope note: this changes what the testbed measures, not just how. Worth doing before any result from it is written up.
